### PR TITLE
Remove user tenants

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review. @tokle FYI.

This removes all traces of user tenants from our controllers.
Existing deployments (there are none right now) are deactivated,
and applications deleted on restart of the controller, for all
applications under a user tenant.
Existing user tenants are deleted on restart of the controller.
APIs pretends creating user tenants is OK, and that any user
tenant already exists, for the purposes of dashboard compatibility.

### CODE REVIEW

**Review by mpolden — APPROVED** [review]
Great! #springcleaning

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java [hunk 1]
```diff
@@ -130,13 +130,18 @@
             int count = 0;
             for (TenantAndApplicationId id: curator.readApplicationIds()) {
                 lockApplicationIfPresent(id, application -> {
-                    if (id.tenant().value().startsWith("by-"))
-                        application = application.with(DeploymentSpec.empty);
-                    else
+                    if (id.tenant().value().startsWith("by-")) {
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]:
```suggestion
                    if (id.tenant().value().startsWith("by-")) { // TODO jonmv: Remove after run once.
```

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/TenantController.java [hunk 2]
```diff
@@ -47,8 +46,12 @@ public TenantController(Controller controller, CuratorDb curator, AccessControl
             Instant start = controller.clock().instant();
             int count = 0;
             for (TenantName name : curator.readTenantNames()) {
-                lockIfPresent(name, LockedTenant.class, this::store);
-                count++;
+                if (name.value().startsWith(Tenant.userPrefix))
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/TenantController.java]:
```suggestion
                if (name.value().startsWith(Tenant.userPrefix)) // TODO jonmv: Remove after run once.

```

**Comment by jonmv** [comment]
Will try in CD tomorrow morning.

**Pull request merged**